pkg/rancher/rancher.go

```diff
@@ -263,3 +279,62 @@ func localClusterEnabled(opts *Options) bool {
 	}
 	return false
 }
+
+// migrateEncryptionConfig uses the dynamic client to get all clusters and then marshals them through the
+// standard go JSON package using the updated backing structs in RKE that include JSON tags. The k8s JSON
+// tools are strict with casing so the fields would be dropped before getting saved back in the proper casing
+// if any controller touches the cluster first. See https://github.com/rancher/rancher/issues/31385
+func migrateEncryptionConfig(ctx context.Context, restConfig *rest.Config) error {
+	dynamicClient, err := k8dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	clusterDynamicClient := dynamicClient.Resource(mgmntv3.ClusterGroupVersionResource)
+
+	clusters, err := clusterDynamicClient.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if !k8serror.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	var allErrors error
+
+	for _, rawDynamicCluster := range clusters.Items {
+		annotations := rawDynamicCluster.GetAnnotations()
+		if annotations != nil && annotations[encryptionConfigUpdate] == "true" {
+			continue
+		}
+
+		clusterBytes, err := rawDynamicCluster.MarshalJSON()
+		if err != nil {
+			allErrors = multierror.Append(errors.Wrap(err, "error trying to Marshal dynamic cluster"), allErrors)
+			continue
+		}
+
+		var cluster *v3.Cluster
+
+		if err := json.Unmarshal(clusterBytes, &cluster); err != nil {
+			allErrors = multierror.Append(errors.Wrap(err, "error trying to Unmarshal dynamicCluster into v3 cluster"), allErrors)
+			continue
+		}
+
+		if cluster.Annotations == nil {
+			cluster.Annotations = make(map[string]string)
+		}
+		cluster.Annotations[encryptionConfigUpdate] = "true"
+
+		u, err := unstructured.ToUnstructured(cluster)
+		if err != nil {
+			allErrors = multierror.Append(err, allErrors)
+			continue
+		}
+
+		_, err = clusterDynamicClient.Update(ctx, u, metav1.UpdateOptions{})
```

Comment:
probably need backoff/retry on this.. maybe ignore errors until some reasonable amount of retries and error out only at the end?

Reply:
Added a retry loop